astrometry/src/PixelMap.cpp

```diff
@@ -350,3 +350,30 @@ ConstantMap::create(const YAML::Node& node,
   }
 }
 #endif
+
+PixelMap*
+ASTMap::duplicate() const {
+  std::shared_ptr<ast::Mapping> newMapping = mapping.copy();
+  return new ASTMap(*newMapping);
+}
+
+void 
+ASTMap::toWorld(double xpix, double ypix,
+		 double& xworld, double& yworld,
+     double color) const {
+
+  std::vector<double> pixCoords{xpix, ypix};
+  std::vector<double> worldCoord = mapping.applyForward(pixCoords);
```

Comment:
I'm a bit worried that this is going to be slow. AST is much faster if you give it many coordinates to transform and it's even faster if you ask it to transform a regular pixel grid to WCS in one go (astMapGrid).

Reply:
Ok. I assume it's not possible to run this in bulk given the way gbdes calls the transformation code, but I imagine it would pull all that time back if you did that.